cvat-core/src/labels.ts

```diff
@@ -98,6 +98,7 @@ export class Label {
     public deleted: boolean;
     public patched: boolean;
     public readonly hasParent?: boolean;
+    public highlight = false;
```

Comment:
While this approach works for highlighting, its not the best way to implement it. The `Label` instance in `cvat-core` is a model class, which basically means that it represents data. It shouldnt know anything about its visual representation(the exception here is color as its shoud be persisted and it comes with data from the server). The highlight state is temporary, its not connected to the server, its a `cvat-ui` feature to improve users ux.
So, lets try to go without model modifications. The highlighted conflict object is stored in `annotations.highlightedConflict` reducer. You can access this data from `frame-tags` component using `mapStateToProps`. Using this data you can check if the tag is highlighed and re-render it with appropriate styles

Reply:
Hello @klakhov , Sorry for the delayed response. While I was going through this, I made appropriate modifications to access this directly from `frame-tags` component. But could you please suggest to me how I should pass this also to [`onUpdateAnnotation`](https://github.com/Viditagarwal7479/cvat/blob/dc94e1fe4e8b2ed13013d3456d2379d0ec679d33/cvat-ui/src/components/annotation-page/canvas/views/canvas2d/canvas-wrapper.tsx#L271-L273) function, because as of now when I call this function directly to highlight tags, the `frame-tags` receives the last highlighted shape type annotations?

Reply:
With the suggested approach(do not add a new field `highlight` to the label model) why do we even need to call onUpdateAnnotation? We shouldnt update annotations in any way while highlighting something.
The idea here just to skip canvasInstance.highlight call for tags and then re-render the `frame-tags` component based on `highlightedConflict` in the state.

Maybe you can commit your changes so I can suggest something more explicitly?

Reply:
Thank you for your response. In the changes that I have pushed just now, I have made appropriate changes. Notably, I have added a function which updates the state value of `highlightedConflict` based on its props value.

But in this, I had to add `// eslint-disable-next-line react/no-unused-state` otherwise, it raised the warning of `unused-state`. I found a similar issue https://github.com/jsx-eslint/eslint-plugin-react/issues/1964 on `jsx-eslint` repo, but it wasn't of much help.